### PR TITLE
Bug/6915 fix ga4 tracking exclusion

### PR DESCRIPTION
**Pull request by derweili**
## Summary

Addresses issue:

- #6915

## Relevant technical choices

Followed IB, not deviations from the IB.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @derweili – this looks good and works well but is missing test coverage.

The IB calls for updating `test_tracking_disabled` in the `AnalyticsTest` but I think this would make more sense to add to the `Analytics_4Test` class. We have a few similar tests in there for the opt-out snippet already so let's add the GA4-specific tests there as would be expected. Looks like you already have some familiarity with this part from a while back but let me know if you have any questions.

File: includes/Modules/Analytics_4.php [hunk 1]
```diff
@@ -122,6 +122,8 @@ function( $old_value, $new_value ) {
 				$this->get_method_proxy( 'replicate_analytics_sharing_settings' )
 			);
 		}
+
+		add_action( 'googlesitekit_allow_tracking_disabled', $this->get_method_proxy( 'filter_analytics_allow_tracking_disabled' ) );
```
Comment (techanvil) [inline includes/Modules/Analytics_4.php:126]: It's a small detail, but this should be `add_filter`, not `add_action`.

```suggestion
		add_filter( 'googlesitekit_allow_tracking_disabled', $this->get_method_proxy( 'filter_analytics_allow_tracking_disabled' ) );
```

**Comment by derweili** [comment]
@aaemnnosttv I added test to the `Analytics_4Test` as you recommended.

However, I also added a test case to the `test_tracking_disabled` in `AnalyticsTest` to handle the case where Analytics is enabled, but not UA property is configured. In that case we should verify the `googlesitekit_analytics_tracking_opt_out` action is still called.

File: tests/phpunit/integration/Modules/AnalyticsTest.php [hunk 2]
```diff
@@ -517,8 +517,10 @@ public function test_tracking_disabled( $settings, $logged_in, $is_tracking_acti
 			// When tracking is active, the `googlesitekit_analytics_tracking_opt_out` action should not be called.
 			$this->assertEquals( 0, did_action( 'googlesitekit_analytics_tracking_opt_out' ) );
 		} else {
-			// When tracking is disabled, the opt out snippet should be present.
-			$this->assertStringContainsString( 'window["ga-disable-UA-21234567-8"] = true', $head_html );
+			if ( ! empty( $settings['propertyID'] ) ) {
+				// When tracking is disabled, the opt out snippet should be present.
+				$this->assertStringContainsString( 'window["ga-disable-UA-21234567-8"] = true', $head_html );
+			}
```
Comment (aaemnnosttv) [inline tests/phpunit/integration/Modules/AnalyticsTest.php:523]: Perhaps here we should add an assertion that it does not contain `window["ga-disable-"]`  when there is no propertyID?

**Review by aaemnnosttv — APPROVED** [review]
Great, thanks @derweili !

**Pull request merged**